desktop/src/main/java/bisq/desktop/main/portfolio/PortfolioView.java

```diff
@@ -228,11 +233,18 @@ private void loadView(Class<? extends View> viewClass, @Nullable Object data) {
                 selectOpenOffersView((OpenOffersView) view);
             }
         } else if (view instanceof DuplicateOfferView) {
-            if (duplicateOfferView == null && data instanceof OfferPayload) {
+            if (duplicateOfferView == null && data instanceof OfferPayloadBase) {
+                // Switch to create BSQ swap offer
+                if (data instanceof BsqSwapOfferPayload) {
```

Comment:
I think it would be better to make that distinction in the caller directly (or a util if it would lead to code duplication). Otherwise its a bit confusing to expect first the DuplicateOfferView which does not support BSQ swaps and later find here the hack to rearrange the navigation.

Reply:
I didn't want to change too much to prevent breaking the duplicate offer calls on multiple places, but yes I could move it into a util so we are not adding lots of duplications in the different caller views.

Reply:
Moved it into a Util